Assets/VRTK/Scripts/Locomotion/VRTK_SlingshotJump.cs

```diff
@@ -0,0 +1,239 @@
+﻿// Player Climb|Locomotion|20120
+namespace VRTK
+{
+    using System;
+    using GrabAttachMechanics;
+    using UnityEngine;
+
+    /// <summary>
+    /// Event Payload
+    /// </summary>
+    /// <param name="controllerIndex">The index of the controller doing the interaction.</param>
+    /// <param name="target">The GameObject of the interactable object that is being interacted with by the controller.</param>
+    public struct SlingshotJumpEventArgs
+    {
+        public GameObject target;
+    }
+
+    /// <summary>
+    /// Event Payload
+    /// </summary>
+    /// <param name="sender">this object</param>
+    /// <param name="e"><see cref="PlayerClimbEventArgs"/></param>
+    public delegate void SlingshotJumpEventHandler(object sender, SlingshotJumpEventArgs e);
+
+    /// <summary>
+    /// Slingshot Jump allows player jumping based on the direction and amount pulled back of each controller. This slingshots the player in defined direction and speed.
+    /// </summary>
+    /// <example>
+    /// `VRTK/Examples/037_CameraRig_ClimbingFalling` shows how to set up a scene with slingshot jumping. This script just needs to be added to the PlayArea object and the requested forces and buttons set.
+    /// </example>
+    [AddComponentMenu("VRTK/Scripts/Locomotion/VRTK_SlingshotJump")]
+    [RequireComponent(typeof(VRTK_BodyPhysics))]
+    public class VRTK_SlingshotJump : MonoBehaviour
+    {
+        [Tooltip("This button will cancel an already tensioned sling shot.")]
+        public VRTK_ControllerEvents.ButtonAlias cancelButton;
```

Comment:
default this to `VRTK_ControllerEvents.ButtonAlias.Undefined` otherwise it will be uninitalised and probably break when scene changing or in a build or whenever unity likes to burp over stupid things